shapash/explainer/smart_plotter.py

```diff
@@ -2817,3 +2834,166 @@ def stability_plot(self,
+        elif isinstance(selection, list):
+            if len(selection) > max_points:
+                print(f"Size of selection is bigger than max_points (default: {max_points}).\
+                      Computation time might be affected")
+            self.explainer.compute_features_compacity(selection, distance, nb_features)
+            self.last_compacity_selection = True
+        else:
+            raise ValueError('Parameter selection must be a list')
+
+        features_needed = self.explainer.features_compacity["features_needed"]
+        distance_reached = self.explainer.features_compacity["distance_reached"]
+
+        # Make plots
+        fig = make_subplots(
+            rows=1,
+            cols=2,
+            subplot_titles=[
+                "Number of features required<br>to explain "
+                + str(round(100 * (1 - distance)))
+                + "% of the model's output",
+                "Percentage of the model output<br>explained by the "
+                + str(nb_features)
+                + " most important<br>features per instance",
+            ],
+            horizontal_spacing=0.2
+        )
+
+        # Used as titles in make_subplots are considered annotations
+        fig.update_annotations(font=self.dict_title_compacity["font"])
+
+        # First plot: number of features required for a given approximation
+        fig.add_trace(
+            go.Histogram(
+                x=features_needed,
+                histnorm="percent",
+                cumulative={"enabled": True},
+                name="",
+                hovertemplate="Top %{x:.0f} features explain "
+                + str(round(100 * (1 - distance)))
+                + "% of the model<br>for %{y:.1f}% of the instances",
+                hovertext="none",
+                marker_color=self.dict_compacity_bar_colors[1],
+                ),
+            row=1,
+            col=1,
+        )
+
+        dict_xaxis = copy.deepcopy(self.dict_xaxis)
+        dict_yaxis = copy.deepcopy(self.dict_yaxis)
+        dict_xaxis['text'] = "Number of selected features"
+        dict_yaxis['text'] = "Cumulative distribution over<br>dataset's instances (%)"
+
+        fig.update_xaxes(title=dict_xaxis, row=1, col=1)
+        fig.update_yaxes(title=dict_yaxis, row=1, col=1)
+
+        # Second plot: approximation reached for a given number of features
+        fig.add_trace(
+            go.Histogram(
+                x=100 * (1 - distance_reached),
+                histnorm="percent",
+                cumulative={"enabled": True, "direction": "decreasing"},
+                name="",
+                hovertemplate="%{x:.0f}% of the output is explained by top "
+                + str(nb_features)
+                + " features<br>for %{y:.1f}% of the instances",
+                marker_color=self.dict_compacity_bar_colors[0],
+                ),
+            row=1,
+            col=2,
+        )
+
+        dict_xaxis = copy.deepcopy(self.dict_xaxis)
```

Comment:
you should rename dict_xaxis which is already used in the first subplot:
dict_xaxis --> dict_xaxis2
dict_yaxis --> dict_yaxis2